### ISSUE TITLE
deferToThread w/ trial -u times out on second pass

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2448]
|<img alt="mwl's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| mwl reported|
|-|-|
|Trac ID|trac#2448|
|Type|defect|
|Created|2007-02-12 09:01:37Z|

Testing code containing deferToThread fails (by timing out) on the second pass when running with trial -u

This code demonstrates the problem:
```
from twisted.trial import unittest
from twisted.internet.threads import deferToThread
 
from time import sleep
 
def blockingFunc():
     sleep(0.1)

class DeferToThreadTestCase(unittest.TestCase):

     def testDeferToThread(self):
         return deferToThread(blockingFunc)
```

[Link to original discussion on Twisted-Python](http://twistedmatrix.com/pipermail/twisted-python/2007-February/014825.html)

<details><summary>Searchable metadata</summary>

```
trac-id__2448 2448
type__defect defect
reporter__mwl mwl
priority__normal normal
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__trial trial
keywords__ 
time__1171270897000000 1171270897000000
changetime__1178617547000000 1178617547000000
version__None None
owner__ 
cc__therve
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_1"></a>|@therve commented|
|-|-|

[#2568](https://github.com/twisted/twisted/issues/2568) was a duplicate of this one.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_3"></a>|@therve set status to `closed`|
|-|-|

It has been solved bu [#1832](https://github.com/twisted/twisted/issues/1832) in r20103.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
